hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/CSMMetrics.java

```diff
@@ -216,6 +244,14 @@ public void incNumEvictedCacheCount() {
     numEvictedCacheCount.incr();
   }
 
+  public void incPendingApplyTransactions() {
+    pendingApplyTransactions.incr();
+  }
+
+  public void decPendingApplyTransactions() {
```

Comment:
What if the transaction fails after `incPendingApplyTransactions` ? Will this cause a “metric leak”?

Reply:
this is called by whenComplete(), which is invoked even if there's an exception.